docs/book/src/developer/testing.md

```diff
@@ -37,7 +37,20 @@ a fake infrastructure provider to allow test coverage for testing the interactio
 
 ## Running unit and integration tests
 
-Using the `test` target through `make` will run all of the unit and integration tests.
+Run `make test` to execute all unit and integration tests.
+
+<aside class="note">
+
+<h1>Tips</h1>
+
+When testing individual packages, you can speed up the test execution by running the tests with a local kind cluster.
+This avoids spinning up a testenv with each test execution. It also makes it easier to debug, because it's straightforward
+to access a kind cluster with kubectl during test execution. For further instructions, run: `./hack/setup-envtest-with-kind.sh`.
+
+When testing individual tests, which don't require testenv (i.e. they are only using fake client), you can skip the testenv 
```

Comment:
This is maybe a bit unclear for new users - it's hard to see which tests use testenv and which don't (except that they fail when this is set). Is there any way we can make this more clear for users? Or will it always be hit and miss unless you're familiar with the specifics of a test?

Reply:
I think this tip is more targeted at advanced users

Reply:
Can we log a warning in test main? Just as a pointer for anyone who might not understand the error? (not on this PR though!)

Reply:
Puh, not sure anyone would find the warning between all the output we already have.